core/src/main/scala/kafka/log/ProducerStateManager.scala

```
@@ -445,8 +445,9 @@ class ProducerStateManager(val topicPartition: TopicPartition,
             lastMapOffset = lastSnapOffset
             return
           } catch {
-            case e: CorruptSnapshotException =>
-              error(s"Snapshot file at ${file.getPath} is corrupt: ${e.getMessage}")
+            case e: Exception =>
```

Aren't there a set of exceptions that are expected?

---

I guess so.. Seems like `IOException`, `SchemaException`, and `CorruptSnapshotException`. We also throw `IllegalArgumentException` if the version is wrong, but I almost think `CorruptSnapshotException` is more appropriate for that case. Since we ignore these errors anyway, I thought we may as well just catch `Exception`.

---

Wouldn't it be better to catch all exceptions here, as the patch does right now? If we can't read the file for any reason, best to delete it and either fallback on the next snaphsot or rebuild the producer state from the log. It seems like the safest approach here.

---

The idea is that we'd use an `error` log for unexpected exceptions.

---

OK. I think that makes sense.

---

Btw, I also agree that `CorruptSnapshotException` would be more appropriate if the version is wrong.

---

Actually, it seems like `readSnapshot` should do much or all of the exception translation, no?

---

Yes, I was thinking about that too.

---

Thinking about this a little bit more, I'm not sure it's a good idea to just catch all exceptions blindly and try to continue. For example, if it is an IOException, there could be a more severe problem. Maybe we should content ourselves with catching only invalid schemas and checksum failures?